src/entity/User.ts

```diff
@@ -8,4 +10,33 @@ export class User extends PickleUserBaseEntity {
 
   @Column()
   password!: string;
+
+  @BeforeInsert()
+  async bI() {
```

Comment:
I would prefer a more descriptive method name here. `hashPasswordBeforeInsert`. I don't think it would be redundant even with the decorator above.

Reply:
I did it this way cause I think those are boilerplate names, I will add email verification under this method so it won't be just password.

Reply:
I am not sure if awaiting a long process like email verification would be a good practice. we might get away with `verified` flag for email.

about the function naming: I think we need a better name too however I am not confident that hashPasswordBeforeInsert would describe it totally. how about `userPreRegistrationHook`?

Reply:
> I am not sure if awaiting a long process like email verification would be a good practice. we might get away with `verified` flag for email.

I'm not sure I follow this. I'm also not sure what @sonercirit means with verification as well. Triggering the verification e-mail (not necessarily sending it right away but adding it to a job queue) here would be OK in my book but I'm willing to listen to objections. Perhaps he meant email input string validation, but I perhaps not.

> about the function naming: I think we need a better name too however I am not confident that hashPasswordBeforeInsert would describe it totally. how about `userPreRegistrationHook`?

this is not just preregistration, it happens on every update. See the latest commit.

Reply:
I meant input string validation, IMO v 0.1 should not have email verification ( you go and click the link in your inbox ), we can add that feature later in the process.

Reply:
But that input string validation will also be moved to apollo layer after moving length checks out